src/Workspaces/CoreTest/Collections/SegmentedArrayTests.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Collections
+{
+    public class SegmentedArrayTests
+    {
+        public static IEnumerable<object[]> TestLengths
+        {
+            get
+            {
+                yield return new object[] { 1 };
+                yield return new object[] { 10 };
+                yield return new object[] { 100 };
+                yield return new object[] { 100000 };
+            }
+        }
+
+        private static void ResetToSequence(SegmentedArray<IntPtr> array)
```

Comment:
why intptr?

Reply:
It has a different size on the different builds.